### PR TITLE
fix: resolves issue with treenav in production

### PR DESCRIPTION
**Pull request by davidmenendez**
fixes issue where treenav isn't responding correctly to url path in production

### CODE REVIEW

File: packages/gatsby-theme-carbon/src/components/LeftNav/LeftNavTree.js [hunk 1]
```diff
@@ -66,7 +66,10 @@ const LeftNavTree = ({ items, theme }) => {
   useEffect(() => {
     const stripTrailingSlash = (str) =>
       str.endsWith('/') ? str.slice(0, -1) : str;
-    setActivePath(stripTrailingSlash(location.pathname));
+    const base = process.env.PATH_PREFIX
+      ? location.pathname.split(`${process.env.PATH_PREFIX}`)[1]
```
Comment (matthewgallo) [inline packages/gatsby-theme-carbon/src/components/LeftNav/LeftNavTree.js:70]:
```suggestion
      ? location.pathname.split(process.env.PATH_PREFIX)[1]
```

Comment (davidmenendez) [reply packages/gatsby-theme-carbon/src/components/LeftNav/LeftNavTree.js:70]: ooops duh 😅

**Review by matthewgallo — APPROVED** [review]
🎉

**Review by szinta — APPROVED** [review]

**Comment by szinta** [comment]
@davidmenendez  I've identified an issue with this PR, the environment variable `process.env.PATH_PREFIX` is specific to [carbon-for-products-website](https://github.com/carbon-design-system/carbon-for-products-website/blob/63cb60ef6830de98f7402b1948ab25760a92d4b8/gatsby-config.js#L7). This means that the fix will only apply to that website or other websites with a similar configuration for setting the `pathPrefix`.

I’ll create another PR with a more generic solution. The current fix will work for carbon-for-products-website, but instead of retrieving the pathPrefix from an environment variable, we can use a GraphQL query to fetch it. This will ensure that the fix works for all Gatsby apps, regardless of the environment variable they use to set pathPrefix.

**Comment by davidmenendez** [comment]
closing in favor of the solution submitted here https://github.com/carbon-design-system/gatsby-theme-carbon/pull/1509

**Pull request closed without merging**